src/PackageCompiler.jl

```diff
@@ -1363,6 +1365,38 @@ function bundle_julia_libraries(dest_dir, stdlibs)
     return
 end
 
+function bundle_julia_libexec(ctx, destdir)
+    # We only bundle the `7z` executable at the moment, which in turn is only needed by Pkg
+    @assert ctx.env.manifest !== nothing
+    has_pkg = false
+    for (_, package) in ctx.env.manifest
+        if package.name == "Pkg"
+            has_pkg = true
+            break
+        end
+    end
+    if !has_pkg
+        return
+    end
```

Comment:
I think a better check would be if `p7zip_jll` is in the manifest, [which Pkg depends on](https://github.com/JuliaLang/Pkg.jl/blob/487f9e271d7ed48eb336409879e793149977d5a0/Project.toml#L25).

Reply:
Sounds good!

Reply:
Done in c87edb8.